src/plugins/vis_types/timeseries/public/trigger_action/metrics_helpers.ts

```diff
@@ -0,0 +1,271 @@
+  return percentiles?.map((percentile) => {
+    return {
+      agg: 'percentile',
+      isFullReference: false,
+      color: percentile.color,
+      fieldName: fieldName ?? 'document',
+      params: { percentile: percentile.value },
+    };
+  });
+};
+
+export const getFormulaSeries = (script: string) => {
+  return [
+    {
+      agg: 'formula',
+      isFullReference: true,
+      fieldName: 'document',
+      params: { formula: script },
+    },
+  ];
+};
+
+export const getPipelineAgg = (subFunctionMetric: Metric) => {
+  const pipelineAggMap = SUPPORTED_METRICS[subFunctionMetric.type];
+  if (!pipelineAggMap) {
+    return null;
+  }
+  return pipelineAggMap.name;
+};
+
+export const getTimeScale = (metric: Metric) => {
+  const supportedTimeScales = ['1s', '1m', '1h', '1d'];
+  let timeScale;
+  if (metric.unit && supportedTimeScales.includes(metric.unit)) {
+    timeScale = metric.unit.replace('1', '');
+  }
+  return timeScale;
+};
+
+export const computeParentSeries = (
+  aggregation: MetricType,
+  currentMetric: Metric,
+  subFunctionMetric: Metric,
+  pipelineAgg: string,
+  meta?: number
+) => {
+  const aggregationMap = SUPPORTED_METRICS[aggregation];
+  const timeScale = getTimeScale(currentMetric);
+  return [
+    {
+      agg: aggregationMap.name,
+      isFullReference: aggregationMap.isFullReference,
+      pipelineAggType: pipelineAgg,
+      fieldName:
+        subFunctionMetric?.field && pipelineAgg !== 'count' ? subFunctionMetric?.field : 'document',
+      params: {
+        ...(currentMetric.window && { window: currentMetric.window }),
+        ...(timeScale && { timeScale }),
+        ...(pipelineAgg === 'percentile' && meta && { percentile: meta }),
+      },
+    },
+  ];
+};
+
+export const getParentPipelineSeries = (
+  aggregation: MetricType,
+  currentMetricIdx: number,
+  metrics: Metric[]
+) => {
+  const currentMetric = metrics[currentMetricIdx];
+  //  percentile value is derived from the field Id. It has the format xxx-xxx-xxx-xxx[percentile]
+  const [fieldId, meta] = currentMetric?.field?.split('[') ?? [];
+  const subFunctionMetric = metrics.find((metric) => metric.id === fieldId);
+  if (!subFunctionMetric) {
+    return null;
+  }
+  const pipelineAgg = getPipelineAgg(subFunctionMetric);
+  if (!pipelineAgg) {
+    return null;
+  }
+  const metaValue = Number(meta?.replace(']', ''));
+  const subMetricField = subFunctionMetric.field;
+  const [nestedFieldId, _] = subMetricField?.split('[') ?? [];
+  // support nested aggs with formula
+  const additionalSubFunction = metrics.find((metric) => metric.id === nestedFieldId);
+  if (additionalSubFunction) {
+    const formula = getParentPipelineSeriesFormula(
+      metrics,
+      subFunctionMetric,
+      pipelineAgg,
+      aggregation,
+      metaValue
+    );
+    if (!formula) {
+      return null;
+    }
+    return getFormulaSeries(formula);
+  } else {
```

Comment:
not: you can remove the `else` branch here as it is returning in all other cases.